validation/shared/src/main/scala/hmda/validation/engine/ts/CommonTsValidation.scala

```diff
@@ -0,0 +1,74 @@
+package hmda.validation.engine.ts
+
+import hmda.model.fi.ts.TransmittalSheet
+import hmda.validation.api.TsValidationApi
+import hmda.validation.dsl.{ Failure, Success }
+import hmda.validation.engine.ValidationError
+import hmda.validation.rules.ts.syntactical.{ S020, S010, S100, S013 }
+import scala.concurrent.{ ExecutionContext, Future }
+import scalaz.Scalaz._
+import scalaz._
+
+trait CommonTsValidation extends TsValidationApi {
+
+  type TsValidation = ValidationNel[ValidationError, TransmittalSheet]
+
+  implicit val ec: ExecutionContext
+
+  protected def s010(t: TransmittalSheet): TsValidation = {
+    S010(t) match {
+      case Success() => t.success
+      case Failure(msg) => ValidationError(s"S010 failed: $msg").failure.toValidationNel
+    }
+  }
+
+  protected def s020(t: TransmittalSheet): TsValidation = {
+    S020(t) match {
+      case Success() => t.success
+      case Failure(msg) => ValidationError(s"S020 failed: $msg").failure.toValidationNel
+    }
+  }
+
+  //TODO: Implement S025 validation rule
+  //  protected def s025(t: TransmittalSheet): TsValidation = {
+  //
+  //  }
+
+  protected def s100(t: TransmittalSheet)(implicit ec: ExecutionContext): Future[TsValidation] = {
```

Comment:
We noticed that `s100` has an implicit `ExecutionContext` and `s013` does not (and seems not to need it, because of the one near the top of the file -- line 16 in this version). Why?